### PR TITLE
Renames to API classes

### PR DESCRIPTION
**Pull request by lukechurch**
@devoncarew PTAL

Do you find these any more explanatory?

Cheers

### CODE REVIEW

File: doc/generated/v1.dart [hunk 1]
```diff
@@ -544,6 +544,36 @@ class AnalysisResults {
 }
 
 
+class CandidateFix {
```
Comment (devoncarew) [inline doc/generated/v1.dart]: This being just `Fix` made sense to me (and I didn't think was likely to conflict with other class names); whichever seems best to you.

File: doc/generated/v1.dart [hunk 2]
```diff
@@ -650,83 +680,52 @@ class DocumentResponse {
...
-
-  Fix.fromJson(core.Map _json) {
-    if (_json.containsKey("edits")) {
-      edits = _json["edits"].map((value) => new Edit.fromJson(value)).toList();
-    }
-    if (_json.containsKey("message")) {
-      message = _json["message"];
-    }
-  }
+class ProblemAndFix {
```
Comment (devoncarew) [inline doc/generated/v1.dart]: `ProblemAndFixes`?

**Comment by devoncarew** [comment]
Thanks for the renames! `Edit` definitely seemed a bit generic.

**Comment by lukechurch** [comment]
@devoncarew Thanks, adjusted as discussed
@kasperpeulen I've put in a flag into the Analysis Result "HasFix". This can be used for setting the wrench icon.

**Pull request merged**